langchain/vectorstores/azuresearch.py

```diff
@@ -195,9 +195,25 @@ def add_texts(
         self,
         texts: Iterable[str],
         metadatas: Optional[List[dict]] = None,
+        extra_fields: Optional[Dict[str, str]] = None,
```

Comment:
is this meant to be  list of dicts (not a dict)?

Reply:
Thanks for your suggestion. I changed it accordingly.

I came across this problem when I wanted to identify the text block using some identifier like _block id_. Initially, my idea was to keep the _extra_ fields as a _list[list[dict]]]_ . 
eg: if there are two blocks of text, the extra fields will be [[{'doc_id':'x'},{block_id':1}],[{'doc_id':'y'},{block_id':2}]]
This will be helpful especially when using question answer with sources, where i can identify the block with a filter

But as you have defined metadatas as a list of dict, I think you have not anticipated this requirement. for me the metadatas was not very useful, as we cannot identify individual blocks with it. Now I have defined extra fields similar to metadatas. But please let me know if you find my previous solution useful. 
If so I should define extra fields as in this commit
#2a0f87e3418997234c99c9f5f5ce077d3e91989f